### ISSUE TITLE
UpdateType type could not be found

### ISSUE THREAD

**Issue by gfreemancz** [malware-dev/MDK-SE#110]
Hello
I folowed your guide, install MDK-SE in to VisualStudion 2019 (15.9.10), created new project for SE script with just Echo("helo world") in Main() function and when try to build/deploy i get following error message:
error CS7069: Reference to type 'UpdateType' claims it is defined in 'Sandbox.Common', but it could not be found

Also when I try to google this type (Sandbox.ModAPI.Ingame.UpdateType) I found it mentioned only in your repository. Did I made some mistake (miss to install/download somethink perhaps?) or is this some bug?

**Comment by malware-dev** [comment]
No, you must have made some kind of mistake in your script - or I would have heard before ;)

**Comment by malware-dev** [comment]
Or the setup

**Comment by gfreemancz** [comment]
Well in the script there is not much space for error,  "Echo("hello world");" is quite simple and all else was generated by VS/MDK...  :-).
Thanks for so fasr response!
Can you think of reason why that "UpdateType" type (i belive it should be some enum type....) is missing? what could i have forget/miss or do wrong? 
BTW, when i comment line/call MDKFactory.Run(program); from "Bootstrapper.cs" (line 24), i can "build" but mainly i can deploy to SE and use my script (and it works ok there...).
In conclusion, with commented line mentioned above, i can work with my scripting, i would just like to know why I get that error... 
Please dont take this as criticism of your work, it is just my professional deformation (automotive industry SW developer) that i need to know what I did wrong :-) 
About my setup, i have clean instalation of Visual Studio 2017 (hell, i just now noticed typo in original messge...) and install of your plugin.

**Comment by malware-dev** [comment]
I am by no way taking this as criticism. It's just that hundreds of people are successfully using this tool every day and this is the first time I've heard of such an issue. The only think I could think of would perhaps be if the references to the game DLLs went wrong somehow. MDK works by directly referencing the game DLLs, and the template used is identical to the one used by the game itself.

**Comment by gfreemancz** [comment]
:-) yes, i do have game installed :-) and there was lot of updatets for SE lately so i was not sure if my problem is because i did somethink wrong or there was change in game that caused my issue. 
My first thought was that some update from Kean caused this, my second was that something in VS installation is wrong (i have not just c# VS installed, i also installed lot of C++ development stuff because i do lot of C++ programing as hoby...) and my experience thaught me that in 80% of issues is more effitient to ask then for several hours/days search by myself :-).
Anyway, thanks of that comment in bootstrapper.cs i can deploy script to SE even if i can´t debug/test it in VS so I can play with ingame programing. 
But I also thought if it is caused by some update of SE you should know about it. Eather way, thanks for your project, it make SE scripting quite enjoyable (at least for freaks like me :-D ).

**Comment by malware-dev** [comment]
It makes _no_ sense though, that commenting out anything in bootstrapper would affect your ability to deploy, There's _no_ connection between that and the error message you posted. None.

I assume you tried a simple Clean/Rebuild?

**Comment by gfreemancz** [comment]
Actualy, there is connection. When i search for definition of MDKFactory::Run(...) method (VS adds "from metha data" comment for that header), I see it has one default parameter of "Sandbox.ModAPI.Ingame.UpdateType" type/enum. and because definition of that type/enum is missing in my solution/inastalation/whatever... VS throw that error. When i comment that method call, all is ok... if I understand your project structurre correctly (and that is a big "if" ) this bootstrapper module is mainly for testing/debugging in VS, is that correct?

**Comment by gfreemancz** [comment]
and yes, I did try "celan/rebuild" :-) for me it has same meaning like "did you try to switch it off and on again" from IT crowd :-D again, almost 10 years of SW developement in automotive :-D that leaves a scars :-D

**Comment by malware-dev** [comment]
Yes, but the very same type is needed in your Main script method for anything but the very simplest of scripts.
